game-core/src/main/java/games/strategy/engine/data/MoveDescription.java

```diff
@@ -0,0 +1,77 @@
+package games.strategy.engine.data;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Maps;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/** Describes an action that moves one or more units along a specific route. */
+public class MoveDescription extends AbstractMoveDescription {
```

Comment:
There is not much in `AbstractMoveDescription`, it might be easier to just have a variable for 
```
  private final Collection<Unit> units;
```
The benefits is we avoid the coupling from inheritance, one of the stronger/strongest forms of coupling out there.

Reply:
Looks like it's used in AbstractUndoableMove.getDescriptionObject(). Not sure its easy to decouple given it's not the only subclass.

Reply:
Is it possible to decouple *just* this one subclass? 

NB: It turns out this update is breaking change to save game serialization. This is a good time to make such changes. After the 2.0 release is launched, and we will need to maintain save-game compatibility.